### ISSUE TITLE
[meta] Improve angr documentation

### ISSUE THREAD

**Issue by ekse** [angr/angr#65]
This is a meta-ticket to keep track of tasks to improve the angr documentation.

- [X] Host the api-doc on the angr website. 
- [X] Automatically rebuild api-doc on commit.
- [X] Disable angr.readthedocs.org
- [X] Configure Gitbook to be generated based on content of angr-doc.
- [X] Add a link to the Gitbook on angr.io
- [X] Add a link to the api-doc on angr.io
- [X] Document how to write docstrings
- [x] Check if Sphinx supports having more than one parameter type. 
- [x] Document how to maintain api-doc

**Comment by rhelmot** [comment]
Could you add me and maybe zardus to the angr gitbook organization you created?

**Comment by rhelmot** [comment]
api-doc is now hosted/linked from angr.io! I tossed Yan an internal pr for angr-doc to rebuild api-doc on each release, which should resolve the second tick. I'll fix up the last two as soon as I can push to the gitbook org :+)

**Comment by ekse** [comment]
Sweet! I added you both as managers to the Gitbook project.

**Comment by rhelmot** [comment]
I made a new version of the gitbook under the org so that we can update it explicitly instead of having it sync with github (the current one is attached to your github and there are some permissions issues), but the settings tab for the new book (and the old book, for that matter) is missing, and trying to manually navigate to that URL nets me a 403. Can you see the settings tabs? Is there a higher level of permission I can be elevated to in order to see settings?

**Comment by ekse** [comment]
I added you as manager for the angr project on gitbook, I don't know why the setting didn't stick the first time around, let me know if it still doesn't work. I renamed my gitbook as angr-staging, the editor is quite covenient to edit the markdown files.

if you want to decide when to have the book updated,  you can link the gitbook with the angr github account but not create a webhook.

**Comment by ekse** [comment]
The gitbook was configured to use http://docs.angr.io/ but this subdomain does not seem to exist, it would need to be added.

**Comment by rhelmot** [comment]
Yeah, that should be happening very very very soon. I don't control the DNS for that domain but someone else who does has been notified that they should do that *now*.

**Comment by rhelmot** [comment]
Alright so: I have angr-doc's webhook set to sync with the staging gitbook. I'll add a line to the release script to push to the release gitbook so it'll always correspond with the version on pip and in the api docs.

One thing I noticed is that we need to make all the links to files that are in the repo but not in the book, e.g. the individual examples, use absolute URLs, since right now they 404 when you click the relative URLs in the book.

Additionally, I went and re-organized the introduction section into multiple parts to make the whole thing more book-shaped. I feel comfortable telling people to use gitbook as the primary means to consume the narrative documentation.
